projects/sienna/index.js

```diff
@@ -0,0 +1,152 @@
+}
+
+async function PairsV2() {
+    const pairs = await queryClient.getContracts(pairCodeIdV2);
+    return pairs.filter((p) => p.label.endsWith(`${factoryContractV2}-${pairCodeIdV2}`));
+}
+
+async function TokenInfo(tokenAddress) {
+    const result = await queryClient.queryContractSmart(tokenAddress, { token_info: {} });
+    return result.token_info;
+}
+
+async function PairsVolumes() {
+    const volumes = []
+
+    const pairs = await Pairs();
+
+    await Promise.all(pairs.map(async contract => {
+        const pair_info = (await queryClient.queryContractSmart(contract.address, "pair_info")).pair_info;
+
+        const token1 = await TokenInfo(pair_info.pair.token_0.custom_token.contract_addr);
+        volumes.push({
+            tokens: new BigNumber(pair_info.amount_0).div(new BigNumber(10).pow(token1.decimals)).toNumber(),
+            symbol: token1.symbol
+        });
+
+        const token2 = await TokenInfo(pair_info.pair.token_1.custom_token.contract_addr);
+        volumes.push({
+            tokens: new BigNumber(pair_info.amount_1).div(new BigNumber(10).pow(token2.decimals)).toNumber(),
+            symbol: token2.symbol
+        });
+    }));
+    return volumes;
+}
+
+async function getLendMarkets() {
+    if (!LEND_OVERSEER_CONTRACT) return [];
+    let markets = [], grabMarkets = true, start = 0;
+
+    while (grabMarkets) {
+        const result = await queryClient.queryContractSmart(LEND_OVERSEER_CONTRACT, {
+            markets: {
+                pagination: {
+                    limit: 10,
+                    start: start
+                }
+            }
+        });
+        if (result && result.entries && result.entries.length) {
+            markets = markets.concat(result.entries);
+            start = markets.length;
+        } else grabMarkets = false;
+    }
+
+    return markets;
+}
+
+async function Lend() {
+    const markets = await getLendMarkets();
+    const block = await queryClient.getHeight();
+    return Promise.all(markets.map(async (market) => {
+        const marketState = await queryClient.queryContractSmart(market.contract.address, {
+            state: {
+                block
+            }
+        });
+        const exchange_rate = await queryClient.queryContractSmart(market.contract.address, {
+            exchange_rate: {
+                block
+            }
+        });
+        const underlying_asset = await queryClient.queryContractSmart(market.contract.address, { underlying_asset: {} });
+        const token = await TokenInfo(underlying_asset.address);
+        return {
+            symbol: token.symbol,
+            tokens_supplied: new BigNumber(marketState.total_supply).times(exchange_rate).div(new BigNumber(10).pow(token.decimals)).toNumber(),
+            tokens_borrowed: new BigNumber(marketState.total_borrows).div(new BigNumber(10).pow(token.decimals).toNumber()).toNumber(),
+        };
+    }));
+}
+
+async function StakedTokens() {
```

Comment:
is this counting staked Sienna LP tokens? If so, we can't count this, because its double counting TVL from PairsVolumes()

Reply:
ahh sorry my mistake - could the TVL from StakedTokens() be removed from the main TVL and exported as 'staking' instead?

Reply:
Yes, removed staked tokens from the tvl and exported it on the staking key